QRHunter/app/src/main/java/com/example/qrhunter/ui/profile/ProfileViewModel.java

```diff
@@ -56,7 +57,7 @@ public void removeScannedQRCode(String qrCodeId, String playerId) {
         // Update this.scannedQRCodes
         ArrayList<QRCode> currentScannedQRCodes = this.scannedQRCodes.getValue();
         currentScannedQRCodes.removeIf(qrCode -> {
-            if (qrCode.getId() == qrCodeId) {
+            if (Objects.equals(qrCode.getId(), qrCodeId)) {
```

Comment:
Was this causing an issue before?

Reply:
nope, i must have changed it when i was trying to figure out what the issue was, it should do the same thing tho.

Reply:
Hmm this is checking whether they both point to the same object reference, i'm not sure if it'll still work

Reply:
i think (qrCode.getId() == qrCodeId) checks if it points to same object reference and object.equals compares the string itself, but i can change it

Reply:
Oh you're right, I got it the other way around mb